File: Algorithms/881. Boats to Save People/kotlin/solution.kt
class Solution {
    fun numRescueBoats(people: IntArray, limit: Int): Int {
        people.sort()
        var i = 0
        var j = people.size - 1
        var ans = 0
        while (i <= j) {
            ans++
            if (people[i] + people[j] <= limit) {
                i++
            }
            j--
        }
        return ans
    }
}

File: Algorithms/881. Boats to Save People/golang/solution.go
func numRescueBoats(people []int, limit int) int {
	sort.Ints(people)
	var ans int
	for i, j := 0, len(people)-1; i <= j; {
		ans++
		if people[i]+people[j] <= limit {
			i++
		}
		j--
	}
	return ans
}
